### PR TITLE
fix(sequential-calls): Use push instead of concat when returning releases

### LINKED ISSUE #1864: When `sequential-calls` is enabled `createReleases` always returns an empty array
#### Environment details

We're using the GitHub action https://github.com/google-github-actions/release-please-action, and have a script that reads the output, see:
https://github.com/cloudquery/cloudquery/blob/0fc7caf321d0ce9513449826158d131d43cac464/.github/workflows/release_pr.yml#L24

#### Steps to reproduce

  1. Enable `sequential-calls` in the manifest releaser configuration
  2. See that the action output is always empty (the cause is release-please code, but the issue is visible in the action)

### PR DESCRIPTION
**Pull request by erezrokah**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1864 🦕
Fixes https://github.com/google-github-actions/release-please-action/issues/546
Also fixes https://github.com/cloudquery/cloudquery/issues/7787

### CODE REVIEW

File: src/manifest.ts [hunk 1]
```diff
@@ -1116,7 +1116,7 @@ export class Manifest {
           releasesByPullRequest[pullNumber],
           pullRequestsByNumber[pullNumber]
         );
-        resultReleases.concat(releases);
+        resultReleases.push(...releases);
```
Comment (erezrokah) [inline src/manifest.ts:1119]: `concat` does not modify the array, it always returns a new one

**Comment by erezrokah** [comment]
Could also fix https://github.com/google-github-actions/release-please-action/issues/546

**Comment by erezrokah** [comment]
Hi lovely release please team 👋 Is there anything missing on my end I should add to this PR?

**Review by chingor13 — APPROVED** [review]

**Comment by erezrokah** [comment]
Thanks for the review, merge and quick release 🚀

**Pull request merged**